src/wrapper/objects/auto_long_array.rs

```diff
@@ -0,0 +1,85 @@
+use std::ptr::NonNull;
+
+use log::debug;
+
+use crate::{errors::*, JNIEnv, objects::JObject};
+use crate::objects::release_mode::ReleaseMode;
+use crate::sys::jlong;
+
+/// Auto-release wrapper for pointer-based long arrays.
+///
+/// This wrapper is used to wrap pointers returned by get_long_array_elements.
+///
+/// These arrays need to be released through a call to release_long_array_elements.
+/// This wrapper provides automatic array release when it goes out of scope.
+pub struct AutoLongArray<'a: 'b, 'b> {
+    obj: JObject<'a>,
+    ptr: NonNull<jlong>,
+    mode: ReleaseMode,
+    is_copy: bool,
+    env: &'b JNIEnv<'a>,
+}
+
+impl<'a, 'b> AutoLongArray<'a, 'b> {
+    /// Creates a new auto-release wrapper for a pointer-based long array
+    ///
+    /// Once this wrapper goes out of scope, `release_long_array_elements` will be
+    /// called on the object. While wrapped, the object can be accessed via
+    /// the `From` impl.
+    pub fn new(
+        env: &'b JNIEnv<'a>,
+        obj: JObject<'a>,
+        ptr: *mut jlong,
+        mode: ReleaseMode,
+        is_copy: bool,
+    ) -> Result<Self> {
+        Ok(AutoLongArray {
+            obj,
+            ptr: NonNull::new(ptr).ok_or_else(|| Error::NullPtr("Non-null ptr expected"))?,
+            mode,
+            is_copy,
+            env,
+        })
+    }
+
+    /// Get a reference to the wrapped pointer
+    pub fn as_ptr(&self) -> *mut jlong {
+        self.ptr.as_ptr()
+    }
+
+    /// Commits the result of the array, if it is a copy
+    pub fn commit(&mut self) {
+        let res = self
+            .env
+            .commit_long_array_elements(*self.obj, unsafe { self.ptr.as_mut() });
+        match res {
+            Ok(()) => {}
+            Err(e) => debug!("error committing long array: {:#?}", e),
```

Comment:
Given it is an unexpected result, I'd use 'error' so that the clients are more likely to see and investigate. `debug` logs are usually disabled for libraries, don't they?